src/box/sql/build.c

```diff
@@ -1194,85 +1196,27 @@ vdbe_emit_fk_constraint_create(struct Parse *parse_context,
 	bool is_alter_add_constr =
 		parse_context->create_table_def.new_space == NULL &&
 		parse_context->create_column_def.space == NULL;
-	if (!is_alter_add_constr) {
-		sqlVdbeAddOp2(vdbe, OP_SCopy, fk->child_id,
-				  constr_tuple_reg + 1);
-	} else {
-		sqlVdbeAddOp2(vdbe, OP_Integer, fk->child_id,
-				  constr_tuple_reg + 1);
-	}
-	if (!is_alter_add_constr && fk_constraint_is_self_referenced(fk)) {
-		sqlVdbeAddOp2(vdbe, OP_SCopy, fk->parent_id,
-				  constr_tuple_reg + 2);
+	if (!is_alter_add_constr)
+		sqlVdbeAddOp2(vdbe, OP_SCopy, fk->child_id, regs);
+	else
+		sqlVdbeAddOp2(vdbe, OP_Integer, fk->child_id, regs);
```

Comment:
I wonder how much simpler the code would be if these plain IDs would be stored into normal registers somewhere before and here we could assume that `fk->child_id` is always a register. Not sometimes a register and other times an ID itself. Not for this patchset, probably.